farm/modeling/adaptive_model.py

```diff
@@ -343,10 +343,11 @@ def logits_to_loss_per_head(self, logits, **kwargs):
         all_losses = []
         for head, logits_for_one_head in zip(self.prediction_heads, logits):
             # check if PredictionHead connected to Processor
-            assert hasattr(head, "label_tensor_name"), \
-                (f"Label_tensor_names are missing inside the {head.task_name} Prediction Head. Did you connect the model"
-                " with the processor through either 'model.connect_heads_with_processor(processor.tasks)'"
-                " or by passing the processor to the Adaptive Model?")
+            if not hasattr(head, "label_tensor_name"):
```

Comment:
Not sure about this one. Can the model work without this? e.g. if it is in inference mode.

If so we should remove the assert, if not and the functionality must break further downstream, then lets keep the assert here